search/management/commands/post_dirty_sounds_to_solr.py

```diff
@@ -18,30 +18,51 @@
 #     See AUTHORS file.
 #
 
+import logging
+
 from django.core.management.base import BaseCommand
+
 from sounds.models import Sound
-from utils.search.search_general import add_all_sounds_to_solr, delete_sound_from_solr, check_if_sound_exists_in_solr
-import logging
+from utils.search.search_general import add_sounds_to_solr, delete_sound_from_solr, check_if_sound_exists_in_solr
+from utils.search.solr import SolrException
+
 logger = logging.getLogger("web")
+console_logger = logging.getLogger("console")
 
 
 class Command(BaseCommand):
     args = ''
     help = 'Add all sounds with index_dirty flag True to SOLR index'
 
     def handle(self, *args, **options):
-        sounds_dirty = Sound.objects.select_related("pack", "user", "license").filter(is_index_dirty=True)
 
         # Index all those which are processed and moderated ok
-        sounds_dirty_to_index = sounds_dirty.filter(moderation_state='OK', processing_state='OK')
+        num_sounds = Sound.objects.filter(processing_state="OK", moderation_state="OK", is_index_dirty=True).count()
         logger.info("Starting posting dirty sounds to solr. %i sounds to be added/updated to the solr index"
-                    % sounds_dirty_to_index.count())
-        num_correctly_indexed_sounds = add_all_sounds_to_solr(sounds_dirty_to_index, mark_index_clean=True)
+                    % num_sounds)
+
+        num_correctly_indexed_sounds = 0
+        slice_size = 1000
+        for i in range(0, num_sounds, slice_size):
+            console_logger.info("Adding %i sounds to solr, slice %i", slice_size, i)
+            try:
+                # Get all sounds moderated and processed ok that has is_index_dirty
+                where = "sound.moderation_state = 'OK' AND sound.processing_state = 'OK' AND is_index_dirty = true " \
```

Comment:
Regarding the `where` clause, the part of `sound.moderation_state = 'OK' AND sound.processing_state = 'OK'` could be part of `Sound.objects.bulk_query_solr` because we're never going to send sounds which are not moderated or processed OK to Solr. Then what you pass in the function would be some sort of "extra where" clause.

Also the condition you use `AND sound.id > %s` is wrong because you are comparing sound IDs with the counter `i`. Sound IDs are not necessarily sequential (even though you sort them) as when sounds are deleted there might be missing IDs. Therefore what happens is that sound IDs grow faster than `i` (`num_sounds` < most recent sound ID). Once the loop is finished, there will still be some sounds not indexed, and also I think we will be sending some sounds twice. I indeed tried that locally and while postgres told me I have 300k sounds moderated and processed ok, after a full reindex I only had 270k in solr.

I understand that you use this strategy to avoid having to retrieve all sound IDs and doing some sort of `WHERE id IN ...`. I see two options:
* Retrieve all sound IDs and iterate them in chunks and do the query. In this case at each iteration we'll be sending 1000 unique sounds to solr.
* Get the id of the most recent sound and use this in the `for i in range(0, num_sounds, slice_size)` instead of `num_sounds`. In this case you won't have to get the IDs but at each iteration we might be sending some sounds that were already sent in the last iteration. 

I guess first option is the best. What do you think @alastair and @andrebola ?

Note that this comments affect both `post_dirty_sounds_to_solr` and `reindex_solr` commands.